website/versioned_docs/version-v3.10.x/mutation-background.md

```diff
@@ -0,0 +1,70 @@
+has one more container to mutate.
+
+The biggest practical issue with this call-in-sequence behavior is latency. Validation webhooks
+(which are called in parallel), have a latency equivalent to the slowest-responding webhook.
+Mutation webhooks have a total latency that is the sum of all mutating webhooks to be called. This
+makes mutation much more latency-sensitive.
+
+This can be particularly harmful for something like external data, where a webhook reaches out to
+a secondary service to gather necessary information. This extra hop can be extra expensive,
+especially if these external calls are not minimized. Gatekeeper translates external data
+references scattered across multiple mutators into a single batched call per external data provider,
+and calls each provider in parallel, minimizing latency.
+
+# Mutation Recursion
+
+Not only are mutators chained, but they recurse as well. This is not only due to Kubernetes'
+[reinvocation policy](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#reinvocation-policy),
+but also due to the nature of the Kubernetes control plane itself, since controllers may modify resources periodically.
+Whether because of the reinvocation policy, or because of control plane behavior, mutators are likely to
+operate on their own output. This has some operational risk. Consider a mutating webhook that prepends a hostname to a docker
+image reference (e.g. prepend `gcr.io/`), if written naievly, each successive mutation would add another prefix, leading to results
+like `gcr.io/gcr.io/gcr.io/my-favorite-image:latest`. Because of this, Kubernetes requires mutation webhooks to be
+[idempotent](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#idempotence).
+
+This is a good idea, but there is one problem: webhooks that are idempotent in isolation may not be idempotent as a group.
+Let's take the above mutator and make it idempotent. We'll give it the following behavior: "if an image reference does
+not start with `gcr.io/`, prepend `gcr.io/`". This makes the webhook idempotent, for sure. But, what if there is another
+team working on the cluster, and they want their own image mutation rule: "if an image reference for the `billing`
+namespace does not start with `billing.company.com/`, prepend `billing.company.com/`". Each of these webhooks would
+be idempotent in isolation, but when chained together you'll see results like
+`billing.company.com/gcr.io/billing.company.com/gcr.io/my-favorite-image:latest`.
+
+At small scales, with small teams, it's relatively easy to ensure that mutations don't interfere with each other,
+but at larger scales, or when multiple non-communicating parties have their own rules that they want to set, it
+can be hard, or impossible to maintain this requirement of "global idempotence".
+
+Gatekeeper attempts to make this easier by designing mutation in such a way that "global idempotence" is an
```

Comment:
Makes sense. 

> Gatekeeper attempts to make this easier by designing mutation in such a way that "global idempotence" is an
emergent property of all mutators, no matter how they are configured.

This reads to me as GK can achieve this so I was thinking it would be useful to show an example of how. If there is no example, then perhaps reword this?

Reply:
I mean, the "how" is the proof doc.

What we can't currently do is modify the container repo for a docker image (though we will be able to once we implement an image mutator).

Reply:
ok sg